kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/DependencyResolver.java

```diff
@@ -80,7 +78,43 @@ public ProxyResolutionResult resolveProxyRefs(KafkaProxy proxy, Context<?> conte
         Objects.requireNonNull(proxy);
         Objects.requireNonNull(context);
         Set<VirtualKafkaCluster> virtualKafkaClusters = context.getSecondaryResources(VirtualKafkaCluster.class);
-        return resolve(virtualKafkaClusters, Set.of(proxy), context);
+        if (virtualKafkaClusters.isEmpty()) {
+            return EMPTY_RESOLUTION_RESULT;
+        }
+
+        CommonDependencies commonDependencies = getCommonDependenciesFrom(context);
```

Comment:
Feels a bit like the API is trying to be a bit too smart wrapping up a slightly arbitrary collection of things, but lets run with a it for a bit longer.

Reply:
yeah, absolutely a bag of bits just to reduce the variables being passed around currently. maybe there is a real concept in there though, we do want access to the resolved dependencies in the end. So maybe it could be like:

step 1. get all the deps we can from the Context
step 2. process the tree of the entity we care about looking for dangling refs, unresolved conditions

Going even further I wonder if it would be cool if the Resolver built up the resolved graph, rather than just returning a map of lookups and making the client call back in to look up the CustomResources. We could return a `ResolvedProxy` containing the `KafkaProxy` and `ResolvedClusters` etc.

Reply:
(also this is a private record for DependencyResolver's internal purposes only)

Reply:
Yeah, but it’s kinda needed for the public api and there wasn’t a diff on the actual api for me to comment on.

something more like a fully resolved graph was more what I was imagining the resolver would do. Anyway changes for another day